src/dovi/rpu_info.rs

```diff
@@ -341,9 +467,77 @@ impl RpusListSummary {
             dm_version_str,
             dm_version_counts,
             l6_meta,
+            l5_str,
+            l8_trims,
+            l9_mdp,
             l1_data,
             l1_stats,
             l2_trims,
+            l2_data: None,
+            l2_stats: None,
         })
     }
+
+    pub fn with_l2_data(rpus: &[DoviRpu]) -> Result<Self> {
+        let mut summary = Self::new(rpus)?;
+
+        let default_l2_for_missing = ExtMetadataBlock::Level2(ExtMetadataBlockLevel2::default());
+
+        let l2_data: Vec<_> = rpus
+            .iter()
+            .map(|rpu| {
+                let block = rpu
+                    .vdr_dm_data
+                    .as_ref()
+                    .and_then(|dm| dm.get_block(2))
+                    .unwrap_or(&default_l2_for_missing);
+
+                if let ExtMetadataBlock::Level2(l2) = block {
+                    (
+                        l2.trim_slope,
+                        l2.trim_offset,
+                        l2.trim_power,
+                        l2.trim_chroma_weight,
+                        l2.trim_saturation_gain,
+                        l2.ms_weight,
+                    )
```

Comment:
I can't find a good reason to transform this into a tuple instead of just storing the L2 struct.